lib/payment_gateways/brain_tree_blue_crypt.rb

```diff
@@ -10,27 +12,34 @@ def initialize(user)
         public_key: payment_gateway_options[:public_key],
         private_key: payment_gateway_options[:private_key]
       )
-      @gateway = @gateway_client.transparent_redirect
+      @gateway = @gateway_client.client_token
+      @customer_gateway = @gateway_client.customer
     end
 
-    def confirm(request)
-      gateway.confirm(request.query_string)
+    def confirm(options, nonce)
+      customer_gateway.update(buyer_reference_for_update, options.deep_merge({payment_method_nonce: nonce, credit_card: { options: {make_default: true, update_existing_token: current_token } }}))
     rescue Braintree::BraintreeError => e
-      notify_exception(e, request.query_string)
+      notify_exception(e, {buyer_reference: buyer_reference_for_update})
```

Comment:
I dunno if we should log all the params sent to 'customer_gateway.update' instead. And if any param is a security issue to be logged, then all the params except for the private ones.

Reply:
well the buyer reference is enough as the error is already indicating what could it be: wrong address, zip code etc ...
What I ned is more information about the account to be updated